src/polymer/analyze-properties.ts

```diff
@@ -150,9 +147,20 @@ export function analyzeProperties(
       prop.readOnly = true;
     }
 
-    prop.type = esutil.CLOSURE_CONSTRUCTOR_MAP.get(prop.type!) || prop.type;
+    // If we've already got a type, then it was found in the jsdocs, and is
+    // canonical. Otherwise we can infer it now from the attribute
+    // deserializer.
+    if (prop.type === undefined && prop.attributeType !== undefined) {
+      prop.type = prop.attributeType;
+      // Map primitive constructors to their primitive value types (e.g. String
+      // => string).
+      prop.type = esutil.CLOSURE_CONSTRUCTOR_MAP.get(prop.type) || prop.type;
+      // Assume this type is possibly null or undefined. Users should add a
+      // @type annotation if they want to tighten this type.
+      prop.type += '|null|undefined';
```

Comment:
What about `' | null | undefined'`

I think the spacing helps legibility.

Reply:
Done.